File: multiply.py
import json
import numpy as np
import time

def handler(request):
    try:
        n = int(request.get("n", 200))

        A = np.random.rand(n, n)
        B = np.random.rand(n, n)

        start = time.time()
        C = np.dot(A, B)
        end = time.time()

        return {
            "statusCode": 200,
            "body": json.dumps({
                "size": n,
                "time": end - start
            })
        }

    except Exception as e:
        return {
            "statusCode": 500,
            "body": str(e)
        }

File: multiply.js
export default function handler(req, res) {
  const n = parseInt(req.query.n || "200");

  // Generate matrices
  const A = Array.from({ length: n }, () =>
    Array.from({ length: n }, () => Math.random())
  );
  const B = Array.from({ length: n }, () =>
    Array.from({ length: n }, () => Math.random())
  );

  const start = Date.now();

  // Matrix multiplication
  const C = Array.from({ length: n }, () => Array(n).fill(0));

  for (let i = 0; i < n; i++) {
    for (let j = 0; j < n; j++) {
      for (let k = 0; k < n; k++) {
        C[i][j] += A[i][k] * B[k][j];
      }
    }
  }

  const end = Date.now();

  res.status(200).json({
    size: n,
    time: (end - start) / 1000
  });
}
